socs/agents/hwp_gripper/agent.py

```diff
@@ -0,0 +1,756 @@
+    pgroup.add_argument('--pru_port', type=int, default=8040,
+                        help='Arbitrary port for actuator encoders')
+    pgroup.add_argument('--control_port', type=int, default=8041,
+                        help='Arbitrary port for actuator control')
+    pgroup.add_argument('--return_port', type=int, default=8042,
+                        help='Arbitrary port for actuator messaging')
+    pgroup.add_argument('--supervisor-id', type=str,
+                        help='Instance ID for HWP Supervisor agent')
+    pgroup.add_argument('--no-data-timeout', type=float, default=45 * 60,
+                        help="Time (sec) after which a 'no_data' action should "
+                        "trigger a shutdown")
+    pgroup.add_argument('--limit_pos', help="Physical position of limit switches (mm)")
+    return parser
+
+
+def main(args=None):
+    parser = make_parser()
+    args = site_config.parse_args(agent_class='HWPGripperAgent',
+                                  parser=parser,
+                                  args=args)
+
+    init_params = {'auto_acquire': True}
+
+    agent, runner = ocs_agent.init_site_agent(args)
+    gripper_agent = GripperAgent(agent, mcu_ip=args.mcu_ip,
+                                 pru_port=args.pru_port,
+                                 control_port=args.control_port,
+                                 return_port=args.return_port,
+                                 supervisor_id=args.supervisor_id,
+                                 no_data_timeout=args.no_data_timeout)
+    agent.register_process('grip_acq', gripper_agent.grip_acq,
+                           gripper_agent._stop_acq)
+    agent.register_process('grip_monitor', gripper_agent.grip_monitor,
+                           gripper_agent._stop_monitor)
+    agent.register_process('grip_collect_pru', gripper_agent.grip_collect_pru,
+                           gripper_agent._stop_collect_pru)
+    agent.register_process('grip_build_pru', gripper_agent.grip_build_pru,
+                           gripper_agent._stop_build_pru)
+    agent.register_task('init_processes', gripper_agent.init_processes,
+                        startup=init_params)
+    agent.register_task('grip_power', gripper_agent.grip_power)
+    agent.register_task('grip_brake', gripper_agent.grip_brake)
+    agent.register_task('grip_move', gripper_agent.grip_move)
+    agent.register_task('grip_home', gripper_agent.grip_home)
+    agent.register_task('grip_inp', gripper_agent.grip_inp)
+    agent.register_task('grip_alarm', gripper_agent.grip_alarm)
+    agent.register_task('grip_reset', gripper_agent.grip_reset)
+    agent.register_task('grip_act', gripper_agent.grip_act)
+    agent.register_task('grip_is_cold', gripper_agent.grip_is_cold)
+    agent.register_task('grip_force', gripper_agent.grip_force)
+    agent.register_task('grip_shutdown', gripper_agent.grip_shutdown)
+    agent.register_task('grip_rev_shutdown', gripper_agent.grip_rev_shutdown)
```

Comment:
Sort of a general comment, but every task/process doesn't need to start with "grip", it will be clear from other context that we're interacting with the gripper agent.